test/framework/integration_framework/fake_peer/fake_peer.hpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INTEGRATION_FRAMEWORK_FAKE_PEER_HPP_
+#define INTEGRATION_FRAMEWORK_FAKE_PEER_HPP_
+
+#include <memory>
+#include <string>
+
+#include <boost/core/noncopyable.hpp>
+#include "framework/integration_framework/fake_peer/yac_network_notifier.hpp"
+#include "interfaces/iroha_internal/abstract_transport_factory.hpp"
+#include "logger/logger.hpp"
+#include "network/impl/async_grpc_client.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    class Keypair;
+  }
+  namespace interface {
+    class CommonObjectsFactory;
+    class Transaction;
+    class TransactionBatchParser;
+    class TransactionBatchFactory;
+  }  // namespace interface
+}  // namespace shared_model
+
+namespace iroha {
+  namespace protocol {
+    class Transaction;
+  }
+  namespace network {
+    class MstTransportGrpc;
+    class MstTransportNotification;
+  }  // namespace network
+  namespace consensus {
+    namespace yac {
+      class NetworkImpl;
+      class YacCryptoProvider;
+      class YacHash;
+    }
+  }
+}  // namespace iroha
+class ServerRunner;
+
+namespace integration_framework {
+
+  class FakePeer final : public boost::noncopyable {
+   public:
+    using TransportFactoryType =
+        shared_model::interface::AbstractTransportFactory<
+            shared_model::interface::Transaction,
+            iroha::protocol::Transaction>;
+
+    FakePeer(
+        const std::string &listen_ip,
+        size_t internal_port,
+        const boost::optional<shared_model::crypto::Keypair> &key,
+        const std::shared_ptr<shared_model::interface::Peer> &real_peer,
+        const std::shared_ptr<shared_model::interface::CommonObjectsFactory>
+            &common_objects_factory,
+        std::shared_ptr<TransportFactoryType> transaction_factory,
+        std::shared_ptr<shared_model::interface::TransactionBatchParser>
+            batch_parser,
+        std::shared_ptr<shared_model::interface::TransactionBatchFactory>
+            transaction_batch_factory,
+        bool agree_all_proposals = true);
+
+    void run();
+
+    void subscribeForMstNotifications(
+        std::shared_ptr<iroha::network::MstTransportNotification> notification);
+
+    std::string getAddress() const;
+
+    const shared_model::crypto::Keypair &getKeypair() const;
+
+    void enableAgreeAllProposals();
+
+    void disableAgreeAllProposals();
+
+    rxcpp::observable<YacNetworkNotifier::StateMessagePtr>
+    get_yac_states_observable();
+
+    void voteForTheSame(
```

Comment:
Add docstrings to all class methods of `FakePeer`.

I agree that some methods are not in such need of a docstring, but all of them should be supplied with meaningful docstring.